detox/test/integration/utils/simplistic-runner.js

```diff
@@ -18,4 +18,7 @@ async function test2() {
   await device.terminateApp();
 }
 
-main().catch((e) => console.error(e));
+main().catch((e) => {
+  console.error(e.stack || e.message || e);
+  process.exit(1);
```

Comment:
As this is expected to be a passing test suite, catching an error must make it exit with a non-zero code.

That way the integration tests can be made stricter.